components/script/dom/bindings/typedarrays.rs

```diff
@@ -4,40 +4,59 @@
 
 #![allow(unsafe_code)]
 
+use std::marker::PhantomData;
 use std::ptr;
 
 use js::jsapi::{Heap, JSObject, JS_GetArrayBufferViewBuffer};
 use js::rust::wrappers::DetachArrayBuffer;
 use js::rust::{CustomAutoRooterGuard, MutableHandleObject};
-use js::typedarray::{
-    CreateWith, Float32Array, JSObjectStorage, TypedArray, TypedArrayElement,
-    TypedArrayElementCreator,
-};
+use js::typedarray::{CreateWith, TypedArray, TypedArrayElement, TypedArrayElementCreator};
 
 use crate::script_runtime::JSContext;
 
-#[derive(Default, JSTraceable)]
-pub struct HeapFloat32Array {
+pub struct HeapTypedArray<T> {
     internal: Box<Heap<*mut JSObject>>,
+    phantom: PhantomData<T>,
 }
 
-impl HeapFloat32Array {
-    pub fn set_data(&self, cx: JSContext, data: &[f32]) -> Result<(), ()> {
+#[allow(unsafe_code)]
```

Comment:
This is unnecessary because the entire module allows unsafe code(see top of file).